stdlib/messaging/rabbitmq/src/main/java/org/ballerinalang/messaging/rabbitmq/nativeimpl/channel/BasicGet.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinalang.messaging.rabbitmq.nativeimpl.channel;
+
+import com.rabbitmq.client.AMQP;
+import com.rabbitmq.client.Channel;
+import com.rabbitmq.client.GetResponse;
+import org.ballerinalang.jvm.BallerinaValues;
+import org.ballerinalang.jvm.scheduling.Strand;
+import org.ballerinalang.jvm.values.ObjectValue;
+import org.ballerinalang.messaging.rabbitmq.RabbitMQConstants;
+import org.ballerinalang.messaging.rabbitmq.RabbitMQUtils;
+import org.ballerinalang.model.types.TypeKind;
+import org.ballerinalang.natives.annotations.BallerinaFunction;
+import org.ballerinalang.natives.annotations.Receiver;
+
+import java.io.IOException;
+
+/**
+ * Retrieves a message from a queue using RabbitMQ pull API.
+ *
+ * @since 1.0.3
+ */
+@BallerinaFunction(
+        orgName = RabbitMQConstants.ORG_NAME,
+        packageName = RabbitMQConstants.RABBITMQ,
+        functionName = "basicGet",
+        receiver = @Receiver(type = TypeKind.OBJECT,
+                structType = RabbitMQConstants.CHANNEL_OBJECT,
+                structPackage = RabbitMQConstants.PACKAGE_RABBITMQ),
+        isPublic = true
+)
+public class BasicGet {
+
+    public static Object basicGet(Strand strand, ObjectValue channelObjectValue, String queueName, Object ackMode) {
+        Channel channel = (Channel) channelObjectValue.getNativeData(RabbitMQConstants.CHANNEL_NATIVE_OBJECT);
+        boolean autoAck = false;
+        if (ackMode.toString().equals("auto")) {
+            autoAck = true;
+        }
+        try {
+            GetResponse response = channel.basicGet(queueName, autoAck);
+            return getMessageObjectValue(response.getBody(), response.getEnvelope().getDeliveryTag(),
+                    response.getProps(), channel, autoAck);
+        } catch (IOException e) {
+            return RabbitMQUtils.returnErrorValue("Error occurred while retrieving the message: " +
+                    e.getMessage());
+        }
+    }
+
+    private static ObjectValue getMessageObjectValue(byte[] message, long deliveryTag, AMQP.BasicProperties properties,
```

Comment:
How about renaming to make the intent clear?
```suggestion
    private static ObjectValue createAndPopulateMessageObjectValue(byte[] message, long deliveryTag, AMQP.BasicProperties properties,
```

Reply:
As this method is a private method used only by this class it is clear what it does in this context, without making the method name unnecessarily long. But if it is still **confusing** we can change the method name.

Reply:
Quoting from clean code:

> Don’t be afraid to make a name long. A long descriptive name is better than a shortenigmatic name. A long descriptive name is better than a long descriptive comment. 

This is true specially for private functions. If you think changing function name doesn't add value we can leave it.

Reply:
Resolving the method name.